kinto/core/resource/__init__.py

```diff
@@ -1139,7 +1164,14 @@ def process_record(self, new, old=None):
         existing ACE is removed (using empty list).
         """
         new = super(ShareableResource, self).process_record(new, old)
-        permissions = self.request.validated.get('permissions', {})
+
+        content_type = str(self.request.headers.get('Content-Type'))
+        # patch is specified as a list of of operations (RFC 6902)
+        if content_type == 'application/json-patch+json':
```

Comment:
since this is repeated several time it could go to a computed `@property` or be initialized as private attribute in constructor (like `if self._is_json_patch`). What do you think?